src/libs/imageManipulator/index.js

```diff
@@ -0,0 +1,143 @@
+import _ from 'underscore';
+
+/**
+ * Calculates a size of canvas after rotation
+ *
+ * @param {Number} width
+ * @param {Number} height
+ * @param {Number} angle
+ * @returns {Object} Returns width and height of new canvas
+ */
+function sizeFromAngle(width, height, angle) {
+    const radians = (angle * Math.PI) / 180;
+    let c = Math.cos(radians);
+    let s = Math.sin(radians);
+    if (s < 0) {
+        s = -s;
+    }
+    if (c < 0) {
+        c = -c;
+    }
+    return {width: (height * s) + (width * c), height: (height * c) + (width * s)};
+}
+
+/**
+ * Creates a new rotated canvas
+ *
+ * @param {Object} canvas
+ * @param {Number} degrees
+ * @returns {Object}
+ */
+function rotateCanvas(canvas, degrees) {
+    const {width, height} = sizeFromAngle(canvas.width, canvas.height, degrees);
+
+    const result = document.createElement('canvas');
```

Comment:
I've never really worked with canvas a whole lot, but why is it necessary to create a new canvas as opposed to just rotating the original one? What happens to the old `canvas`? If it's not destroyed anywhere, is that a memory leak?

Reply:
Unfortunately, it is not possible to implement your suggestion. Primary, because we can't manipulate by already drawn objects on the canvas. We can only apply transformations to future drawings. I've added a comment.

Reply:
What about something like example 1 here: https://stackoverflow.com/a/30250725/9114791